app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -500,18 +520,60 @@ void
 BrowserWorld::SetSurfaceTexture(const std::string& aName, jobject& aSurface) {
   VRB_LOG("SetSurfaceTexture: %s", aName.c_str());
   if (m.env && m.activity && m.dispatchCreateWidgetMethod) {
-    for (WidgetPtr& widget: m.widgets) {
-      if (aName == widget->GetSurfaceTextureName()) {
-        int32_t width = 0, height = 0;
-        widget->GetSurfaceTextureSize(width, height);
-        m.env->CallVoidMethod(m.activity, m.dispatchCreateWidgetMethod, widget->GetType(),
-                              widget->GetHandle(), aSurface, width, height);
-        return;
-      }
+    WidgetPtr widget = m.FindWidget([=](const WidgetPtr & w) {
```

Comment:
Extra space here. Also, I would prefer if we avoid single character var names as it makes it a pain to do search and replace on them (I realize m is a single character but it always has a . next to it :) ).